ipaserver/install/ipa_replica_prepare.py

```diff
@@ -321,9 +320,6 @@ def run(self):
 
         logger.info("Preparing replica for %s from %s",
                     self.replica_fqdn, api.env.host)
-        enable_replication_version_checking(
-            api.env.realm,
-            self.dirman_password)
```

Comment:
The whole point of this is to provide a relief valve in the case where we make some breaking architectural change in IPA so that one version can't replicate with another. If the check is not done at install time this is pointless, right?

Reply:
Currently the mechanism checks at each replication session, that versions are compatible. So it is not only at install. In addition I am not sure this mechanism would prevent install between incompatible versions.
If we want to check compatible version we can also add a version entry/attribute in the DIT instead of breaking replication

Reply:
I don't mean to imply it is only at install but if it isn't checked at install time then the installation could be successful only to fail at the first replication event right?

We don't have to stick with this replication version but it was, as I understood it, the best or only way to do this years ago.

Reply:
according to https://bugzilla.redhat.com/show_bug.cgi?id=1623761#c12, the problem is dynamic and transient. So it is quite possible that even if the problem occurs during install, replication retries and install script does not detect the error. During runtime, retry occurs and replication looks just fine.

Repl_version is a nice piece of code and I think the intention was to prevent incompatible versions to pollute each other. So it is enforcing compatibility in a stronger way than uniq/regular version checking from DIT entry. Now being a plugin there is always the risk that it is disabled on one/both side and version not checked. 

If we want to keep this ability to prevent a new version to break an old one (or the opposite), we should keep this plugin enabled.

Reply:
@rcritten , I have a mixed feeling regarding the repl_version checking.

This plugin was created to prevent incompatible versions to break each other. So it is addressing rare condition that never occurred so far and now we are willing to disable it by default.

Now if we want to keep such safety mechanism . It should be enabled on most or all instances. incompatibility checking requires that the both sides have it enabled.

In short, currently repl_version is disabled on Master and enabled on Replica. This patch is disabling it on Replica. I wonder if in the opposite it should not enable it as well on master.